### PR TITLE
🏗 Replace Google Cloud Storage with CircleCI Artifacts

### PR DESCRIPTION
**Pull request by danielrozenberg**
To be merged and deployed in sync with https://github.com/ampproject/amp-github-apps/pull/1265

This also replaced zip with tar.gz - zip is not supported for streaming and this made it difficult keep it around. .tar.gz files are 1% smaller too 😅

### CODE REVIEW

File: build-system/pr-check/utils.js
Comment (rsimha) [inline build-system/pr-check/utils.js:287]: For self-documentation and clarity, I think the function name should still mention the processing part. How about something like `processAndStoreBuildToArtifacts()`? (Or pick something else you like.)

File: build-system/pr-check/utils.js
Comment (rsimha) [inline build-system/pr-check/utils.js:291]: Since this function should be a no-op if called outside CircleCI, maybe rewrite it as follows for clarity?

```js
async function processAndStoreBuildToArtifacts() {
  if (!isCircleciBuild()) {
    return;
  }
  await replaceUrls('test/manual');
  await replaceUrls('examples');
  // Do other stuff
}
```

File: .circleci/setup_storage.sh
Comment (rsimha) [inline .circleci/setup_storage.sh:41]: I'm particularly glad that this part is going away!

**Review by rsimha — APPROVED** [review]
LGTM with a couple suggestions below. Might be worth getting a second review from Esther in case I missed something.

File: build-system/pr-check/utils.js
Comment (danielrozenberg) [reply build-system/pr-check/utils.js:287]: Done

File: build-system/pr-check/utils.js
Comment (danielrozenberg) [reply build-system/pr-check/utils.js:291]: Done

**Review by estherkim — APPROVED** [review]

**Comment by danielrozenberg** [comment]
@estherkim @rsimha To make the transition easier, I restored the Cloud Storage behaviour (so that _both_ Cloud Storage _and_ CircleCI Artifacts will be used), gated by an env variable.

We can let this PR settle for ~2 weeks, then deploy the changed app and flip the env variable. Thoughts?

**Comment by rsimha** [comment]
> We can let this PR settle for ~2 weeks, then deploy the changed app and flip the env variable. Thoughts?

Not a bad idea! I'm good with switching after a few builds have run using the new flow, and will leave this call to you.

**Pull request merged**
